tensorflow_addons/optimizers/conditional_gradient_test.py

```diff
@@ -0,0 +1,643 @@
+            self.assertAllCloseAccordingToType(
+                        np.array([
+                            (3.0 * 0.5 -                            \
+                             (1 - 0.5) * 0.01 * 0.01 / norm1) * 0.5 \
+                            - (1 - 0.5) * 0.01 * 0.01 / norm1,
+                            (4.0 * 0.5 -                            \
+                             (1 - 0.5) * 0.01 * 0.01 / norm1) * 0.5 \
+                            - (1 - 0.5) * 0.01 * 0.01 / norm1
+                        ]), self.evaluate(var1))
+
+
+    def testBasic(self):
+        with self.cached_session():
+            self.doTestBasic(use_resource=False)
+
+
+    @test_utils.run_in_graph_and_eager_modes(reset_test=True)
+    def testResourceBasic(self):
+        self.doTestBasic(use_resource=True)
+
+    def testBasicCallableParams(self):
+        #with tf.enable_eager_execution():
+        self.doTestBasic(use_resource=True, use_callable_params=True)
+
+    def testVariablesAcrossGraphs(self):
+        optimizer = cg_lib.ConditionalGradientOptimizer(0.01, 0.5)
+        with tf.Graph().as_default():
+            var0 = tf.Variable(
+                        [1.0, 2.0], dtype=tf.float32, name="var0")
+            var1 = tf.Variable(
+                        [3.0, 4.0], dtype=tf.float32, name="var1")
+            loss = lambda: tf.math.reduce_sum(var0 + var1)
+            optimizer.minimize(loss, var_list=[var0, var1])
+            optimizer_variables = optimizer.variables()
+        # There should be three items. The first item is iteration,
+        #and one item for each variable.
+            self.assertStartsWith(optimizer_variables[1].name,
+                        "ConditionalGradient/var0")
+            self.assertStartsWith(optimizer_variables[2].name,
+                        "ConditionalGradient/var1")
+            self.assertEqual(3, len(optimizer_variables))
+        '''
+        with tf.Graph().as_default():
+            var2 = tf.Variable(
+                        [1.0, 2.0], dtype=tf.float32, name="var2")
+            var3 = tf.Variable(
+                        [3.0, 4.0], dtype=tf.float32, name="var3")
+            loss = lambda: tf.math.reduce_sum(var2 + var3)
+            optimizer.minimize(loss, var_list=[var2, var3])
+            optimizer_variables = optimizer.variables()
+            self.assertStartsWith(optimizer_variables[1].name, 
+                        "ConditionalGraident/var2")
+            self.assertStartsWith(optimizer_variables[2].name, 
+                        "ConditionalGraident/var3")
+            self.assertEqual(3, len(optimizer_variables))
+        '''
+
+    @test_utils.run_in_graph_and_eager_modes(reset_test=True)
+    def testMinimizeSparseResourceVariable(self):
+        for dtype in [tf.half, tf.float32, tf.float64]:
+            # This test invokes the ResourceSparseApplyConditionalGradient
+            # operation, which did not have a registered GPU kernel as of
+            # April 2018.
```

Comment:
Is the limitation still an issue? Or can it be placed on GPU now?

Reply:
I think you are right and it should not be an issue here any more. Thank you for pointing it out! I will fix this comment part.